app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageComposerViewModel.kt

```diff
@@ -396,5 +370,35 @@ class MessageComposerViewModel @Inject constructor(
         }
     }
 
-    fun provideTempCachePath(): Path = kaliumFileSystem.rootCachePath
+    @Suppress("MagicNumber")
+    fun attachmentPicked(attachmentUri: UriAsset) = viewModelScope.launch(dispatchers.io()) {
+        val tempCachePath = kaliumFileSystem.rootCachePath
+        val assetBundle = fileManager.getAssetBundleFromUri(attachmentUri.uri, tempCachePath)
+        if (assetBundle != null) {
+            // The max limit for sending assets changes between user and asset types.
+            // Check [GetAssetSizeLimitUseCase] class for more detailed information about the real limits.
+            val maxSizeLimitInBytes = getAssetSizeLimit(isImage = assetBundle.assetType == AttachmentType.IMAGE)
+            if (assetBundle.dataSize <= maxSizeLimitInBytes) {
+                sendAttachmentMessage(assetBundle)
+            } else {
+                if (attachmentUri.saveToDeviceIfInvalid) {
+                    with(assetBundle) { fileManager.saveToExternalMediaStorage(fileName, dataPath, dataSize, mimeType, dispatchers) }
+                }
+                val sizeOf1MB = 1024 * 1024
+                messageComposerViewState = messageComposerViewState.copy(
+                    assetTooLargeDialogState = AssetTooLargeDialogState.Visible(
+                        type = assetBundle.assetType,
+                        maxLimitInMB = maxSizeLimitInBytes.div(sizeOf1MB).toInt(),
+                        savedToDevice = attachmentUri.saveToDeviceIfInvalid
+                    )
+                )
+            }
+        } else {
+            onSnackbarMessage(ConversationSnackbarMessages.ErrorPickingAttachment)
```

Comment:
Can you cover with test the case where `assetBundle` is null and we should show the snackbar ?